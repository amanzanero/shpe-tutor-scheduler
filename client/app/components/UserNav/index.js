/**
 *
 * UserNav
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import mobile from 'is-mobile';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import Divider from '@material-ui/core/Divider';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import MailIcon from '@material-ui/icons/Mail';
import ExitToAppIcon from '@material-ui/icons/ExitToApp';

const styles = {
  root: {
    flexGrow: 1,
  },
  grow: {
    flexGrow: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
};

function UserNav(props) {
  const { classes } = props;
  const isMobile = mobile();
  const [state, setState] = React.useState({
    drawerOpen: false,
  });

  const toggleDrawer = (side, open) => () => {
    setState({ ...state, [side]: open });
  };

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <IconButton
            className={classes.menuButton}
            color="secondary"
            aria-label="Menu"
            onClick={toggleDrawer('drawerOpen', true)}
          >
            <MenuIcon />
          </IconButton>
          <Typography variant="h6" color="secondary" className={classes.grow}>
            {isMobile ? 'SHPE Tutoring' : 'SHPE Tutoring and Academics'}
          </Typography>
          {!isMobile && <Button color="secondary">Login</Button>}
        </Toolbar>
      </AppBar>
      <Drawer
        open={state.drawerOpen}
        onClose={toggleDrawer('drawerOpen', false)}
      >
        <div
          tabIndex={0}
          role="button"
          onClick={toggleDrawer('drawerOpen', false)}
          onKeyDown={toggleDrawer('drawerOpen', false)}
        >
          <div className={classes.list}>
            <List>
              <ListItem button key="messages">
                <ListItemIcon>
                  <MailIcon />
                  <ListItemText primary="Messages" />
                </ListItemIcon>
              </ListItem>
            </List>
            <Divider />
            <List>
              <ListItem button key="Log In">
                <ListItemIcon>
                  <ExitToAppIcon />
                  <ListItemText primary="Log In" />
                </ListItemIcon>
              </ListItem>
            </List>
          </div>
        </div>
      </Drawer>
    </div>
  );
}

UserNav.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(UserNav);
