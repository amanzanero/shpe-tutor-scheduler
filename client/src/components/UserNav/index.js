/**
 *
 * UserNav
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import mobile from 'is-mobile';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import Divider from '@material-ui/core/Divider';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import MailIcon from '@material-ui/icons/Mail';
import ExitToAppIcon from '@material-ui/icons/ExitToApp';

const styles = {
  grow: {
    flexGrow: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
};

class UserNav extends React.Component {
  state = {
    drawerOpen: false,
  };

  toggleDrawer = open => () => {
    this.setState({ drawerOpen: open });
  };

  render() {
    const { classes } = this.props;
    const isMobile = mobile();
    const { drawerOpen } = this.state;
    return (
      <div>
        <AppBar position="static">
          <Toolbar>
            <IconButton
              className={classes.menuButton}
              color="secondary"
              aria-label="Menu"
              onClick={this.toggleDrawer(true)}
              data-testid="menu-button"
            >
              <MenuIcon />
            </IconButton>
            <Typography variant="h6" color="secondary" className={classes.grow}>
              {isMobile ? 'SHPE Tutoring' : 'SHPE Tutoring and Academics'}
            </Typography>
          </Toolbar>
        </AppBar>
        <Drawer open={drawerOpen} onClose={this.toggleDrawer(false)}>
          <div
            tabIndex={0}
            role="button"
            onClick={this.toggleDrawer(false)}
            onKeyDown={this.toggleDrawer(false)}
            data-testid="drawer"
          >
            <div className={classes.list}>
              <List>
                <ListItem button key="messages">
                  <ListItemIcon>
                    <MailIcon />
                  </ListItemIcon>
                  <ListItemText primary="Messages" />
                </ListItem>
                <Divider />
                <ListItem button key="Log In">
                  <ListItemIcon>
                    <ExitToAppIcon />
                  </ListItemIcon>
                  <ListItemText primary="Log In" />
                </ListItem>
              </List>
            </div>
          </div>
        </Drawer>
      </div>
    );
  }
}

UserNav.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default withStyles(styles)(UserNav);
