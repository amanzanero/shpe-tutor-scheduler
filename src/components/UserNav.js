/**
 *
 * UserNav
 *
 */

import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import mobile from 'is-mobile';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import Divider from '@material-ui/core/Divider';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import MailIcon from '@material-ui/icons/Mail';
import ExitToAppIcon from '@material-ui/icons/ExitToApp';
import SettingsIcon from '@material-ui/icons/Settings';
import BookIcon from '@material-ui/icons/Collections';

const styles = theme => ({
  grow: {
    flexGrow: 1,
  },
  text: {
    flexGrow: 1,
    fontSize: 26,
    fontFamily: 'Roboto Slab',
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  margin: {
    margin: '.5em',
  },
  gold: {
    color: theme.palette.secondary.light,
  },
  white: {
    color: 'white',
  },
});

class UserNav extends React.Component {
  state = {
    drawerOpen: false,
  };

  toggleDrawer = open => () => {
    this.setState({ drawerOpen: open });
  };

  render() {
    const {
      classes,
      onToggleModal,
      logOut,
      onToggleAddCourses,
      children,
    } = this.props;
    const isMobile = mobile();
    const text = isMobile ? 'Tutors' : 'Tutoring and Academics';
    const { drawerOpen } = this.state;
    return (
      <Fragment>
        <div>
          <AppBar position="static">
            <Toolbar>
              <IconButton
                className={`${classes.menuButton} ${classes.white}`}
                aria-label="Menu"
                onClick={this.toggleDrawer(true)}
                data-testid="menu-button"
              >
                <MenuIcon />
              </IconButton>
              <Typography
                variant="h6"
                className={`${classes.text} ${classes.white}`}
              >
                <span className={classes.gold}>SHPE </span>
                {text}
              </Typography>
            </Toolbar>
          </AppBar>
          <Drawer open={drawerOpen} onClose={this.toggleDrawer(false)}>
            <div
              tabIndex={0}
              role="button"
              onClick={this.toggleDrawer(false)}
              onKeyDown={this.toggleDrawer(false)}
              data-testid="drawer"
            >
              <div className={classes.list}>
                <List>
                  <ListItem
                    button
                    key="messages"
                    onClick={() => console.log('hi')}
                  >
                    <ListItemIcon>
                      <MailIcon />
                    </ListItemIcon>
                    <ListItemText primary="Messages" />
                  </ListItem>
                  <Divider />
                  <ListItem
                    button
                    key="Settings"
                    onClick={() => onToggleModal()}
                  >
                    <ListItemIcon>
                      <SettingsIcon />
                    </ListItemIcon>
                    <ListItemText primary="Settings" />
                  </ListItem>
                  <ListItem
                    button
                    key="Courses"
                    onClick={() => onToggleAddCourses()}
                  >
                    <ListItemIcon>
                      <BookIcon />
                    </ListItemIcon>
                    <ListItemText primary="Courses" />
                  </ListItem>
                  <ListItem button key="Log Out" onClick={() => logOut()}>
                    <ListItemIcon>
                      <ExitToAppIcon />
                    </ListItemIcon>
                    <ListItemText primary="Log Out" />
                  </ListItem>
                </List>
              </div>
            </div>
          </Drawer>
        </div>
        {children}
      </Fragment>
    );
  }
}

UserNav.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  onToggleModal: PropTypes.func.isRequired,
  logOut: PropTypes.func.isRequired,
};

export default withStyles(styles)(UserNav);
