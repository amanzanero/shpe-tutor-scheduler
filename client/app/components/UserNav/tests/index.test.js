/**
 *
 * Tests for UserNav
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';

import UserNav from '../index';

afterEach(cleanup);

describe('<UserNav />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<UserNav />);
    expect(spy).not.toHaveBeenCalled();
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<UserNav />);
    expect(firstChild).toMatchSnapshot();
  });

  it('Drawer appears after menubutton clicked', () => {
    const { getByTestId } = render(<UserNav />);
    const menuButton = getByTestId('menu-button');
    fireEvent(menuButton, new MouseEvent('click'));
  });
});
