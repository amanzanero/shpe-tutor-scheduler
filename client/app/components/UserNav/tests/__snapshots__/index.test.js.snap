// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserNav /> Should render and match the snapshot 1`] = `
<div
  class="UserNav-root-145"
>
  <header
    class="MuiPaper-root-157 MuiPaper-elevation4-163 MuiAppBar-root-148 MuiAppBar-positionStatic-152 MuiAppBar-colorPrimary-155"
  >
    <div
      class="MuiToolbar-root-184 MuiToolbar-regular-186 MuiToolbar-gutters-185"
    >
      <button
        aria-label="Menu"
        class="MuiButtonBase-root-194 MuiIconButton-root-188 MuiIconButton-colorSecondary-191 UserNav-menuButton-147"
        data-testid="menu-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label-193"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-197"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root-282"
        />
      </button>
      <h6
        class="MuiTypography-root-206 MuiTypography-h6-223 MuiTypography-colorSecondary-237 UserNav-grow-146"
      >
        SHPE Tutoring and Academics
      </h6>
      <button
        class="MuiButtonBase-root-194 MuiButton-root-242 MuiButton-text-244 MuiButton-textSecondary-246 MuiButton-flat-247 MuiButton-flatSecondary-249"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-243"
        >
          Login
        </span>
        <span
          class="MuiTouchRipple-root-282"
        />
      </button>
    </div>
  </header>
</div>
`;
